File: 算法/Python/541. Reverse String II.py
class Solution:
    def reverseStr(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: str
        """
        s = list(s)  # convert str to list
        for i in range(0, len(s), 2*k):  # Traversing s ,step = 2*k
            if i-1 <= 0:
                s[i:i+k] = s[i+k-1::-1]  # reverse (i,i+k),when at beginning
            else:
                s[i:i+k] = s[i+k-1:i-1:-1]  # reverse(i,i+k)
        return ''.join(s)

File: 算法/C++/541. Reverse String II.cpp
class Solution {
public:
    string reverseStr(string s, int k) {

        for (int i = 0; i < s.size(); i += 2 * k) {
            int l = i;
            int r = i + k - 1;
            while (r >= s.size()) r--;
            while (l < r)
                swap(s[l++], s[r--]);
        }
        return s;
    }
};